tests/spec/core/mdn-annotation-spec.js

```diff
@@ -2,17 +2,46 @@ import { flushIframes, makeRSDoc, makeStandardOps } from "../SpecHelper.js";
 
 describe("Core - MDN Annotation", () => {
   afterAll(flushIframes);
+
+  const baseJsonPath = `${window.location.origin}/tests/data/mdn-annotation/`;
   let doc;
   beforeAll(async () => {
     const ops = makeStandardOps({
-      mdn: {
-        baseJsonPath: `${window.location.origin}/tests/data/mdn-annotation/`,
-      },
+      mdn: { baseJsonPath },
       shortName: "payment-request",
     });
     doc = await makeRSDoc(ops, "spec/core/mdn-annotation.html");
   });
 
+  describe("key", () => {
+    const getMdnBox = doc =>
+      doc.getElementById("paymentrequest-interface").previousElementSibling;
+
+    it("does nothing if neither shortName or key are providded", async () => {
+      const ops = makeStandardOps({ mdn: { baseJsonPath }, shortName: null });
+      const doc = await makeRSDoc(ops, "spec/core/mdn-annotation.html");
+      expect(getMdnBox(doc).matches("aside.mdn")).toBeFalse();
+    });
+
+    it("prefers `mdn.key` over shortName", async () => {
+      const ops = makeStandardOps({
+        mdn: { baseJsonPath, key: "payment-request" },
+        shortName: "whatever",
+      });
+      const doc = await makeRSDoc(ops, "spec/core/mdn-annotation.html");
+      expect(getMdnBox(doc).matches("aside.mdn")).toBeTrue();
+    });
+
+    it("supports mdn to be a string", async () => {
```

Comment:
This test sadly fetches over network, as we cannot set `mdn.baseJsonPath` (used for mock JSON source) on a string.
I will add a util for to know if we're in test env for these things in a separate PR.

Reply:
Well, all these tests are going over network right now anyway as tests don't add `mdn.specMapUrl` :man_facepalming:

Reply:
We can live with that...